### DISCUSSION TITLE
How can we update/ delete events in google calendar by first asking email, is there any property

### DISCUSSION THREAD

**Discussion by SanaWaseem** [InteractionDesignFoundation/add-event-to-calendar-docs#50] [General]
Is there a way to update/ delete the google calendar event following is my link 

https://www.google.com/calendar/event?action=TEMPLATE&text=Piercing+Studio+Appointment+(Ossington+Piercing+Studio)&dates=20230716T202000Z/20230716T204000Z&location=64+Ossington+Avenue,+Toronto,+Ontario,+M6J2Y7&details=64+Ossington+Avenue,+Toronto,+Ontario,+M6J2Y7%0A%0AView/Change+Appointment:%0Ahttps://app.acuityscheduling.com/schedule.php?owner%3D28316543%26id%255B%255D%3D87dc83cf48dc52001b1dd5f79ce9b0ef%26action%3Dappt%0A%0A(created+by+Acuity+Scheduling)%0AAcuityID%3D1071322728%0AGoogle&sf=true&output=xml&pli=1

 i have to update the event once added.

**Comment by alies-dev** [comment]
**Update: partly possible. Delete: no.**

Google's deep link parser reads an `eid` parameter, the base64 event id. With it the link opens that existing event in the editor instead of creating a new one, so the user can change and save it. There is no parameter that deletes an event, and nothing in the parser that would perform any destructive action, which makes sense given that a link like this is user facing.

There is also no parameter that looks up an event by attendee email. If you need to find the event first, that is the Calendar API, not a URL.

Two notes on the link in the question:

* `https://www.google.com/calendar/event?action=TEMPLATE&...` is the legacy host. It still redirects, but the current entry point is `https://calendar.google.com/calendar/render?action=TEMPLATE&...`;
* `sf=true`, `output=xml` and `pli` are leftovers from the old calendar. The current client does not read any of them.

For the other services: Yahoo has the same idea with `uid` (plus `recurId` for a single occurrence of a recurring event), which opens an existing event for editing. Outlook's compose deep link has no equivalent.
